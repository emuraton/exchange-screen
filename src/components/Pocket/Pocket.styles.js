import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 100px;
  height: 200px;
`;

export const LeftFlexWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`;

export const RightFlexWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: right;
`;

const CurrencyLabel = styled.span`
  font-size: 2rem;
  font-weight: 600;
`;

CurrencyLabel.displayName = 'CurrencyLabel';
export { CurrencyLabel };

const AmountLabel = styled.span`
  font-size: 1.1rem;
  font-weight: 200;
`;
AmountLabel.displayName = 'AmountLabel';
export { AmountLabel };

const AmountInput = styled.input`
  font-size: 1.1rem;
  font-weight: 300;
  height: 30px;
`;

AmountInput.displayName = 'AmountInput';
export { AmountInput };
