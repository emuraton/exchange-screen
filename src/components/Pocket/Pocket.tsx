import React, { useEffect, useRef } from 'react';

import {
  Container,
  LeftFlexWrapper,
  RightFlexWrapper,
  CurrencyLabel,
  AmountLabel,
  AmountInput
} from './Pocket.styles.js';

type IProps = {
  currency: string;
  currencyLabel: string;
  pocketAmount: number;
  focusOnPageLoading?: boolean;
  onChangeHandler?: Function;
  inputValue?: string;
  index?: string;
};

const Pocket = ({
  currency,
  currencyLabel,
  pocketAmount,
  focusOnPageLoading,
  onChangeHandler,
  inputValue
}: IProps) => {
  const inputEl = useRef<HTMLInputElement>(null);

  // fallback for html5 autofcus
  useEffect(() => {
    if (focusOnPageLoading && inputEl.current) {
      inputEl.current.focus();
    }
  }, [focusOnPageLoading]);

  return (
    <Container>
      <LeftFlexWrapper>
        <CurrencyLabel>{currencyLabel}</CurrencyLabel>
        <AmountLabel>
          You have {currency}
          {pocketAmount}
        </AmountLabel>
      </LeftFlexWrapper>
      <RightFlexWrapper>
        <AmountInput
          type="text"
          autoFocus
          value={inputValue}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            onChangeHandler && onChangeHandler(e.target.value)
          }
          ref={inputEl}
        />
      </RightFlexWrapper>
    </Container>
  );
};

export default Pocket;
