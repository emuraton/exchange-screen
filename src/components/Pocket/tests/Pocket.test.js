import React from 'react';
import { shallow } from 'enzyme';

import Pocket from '../Pocket';

describe('<Pocket />', () => {
  const initialProps = {
    currency: 'EUR',
    currencyLabel: 'Â£',
    pocketAmount: 55.22,
    inputValue: ''
  };

  it('should render correctly', () => {
    const wrapper = shallow(<Pocket {...initialProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the right currency label', () => {
    const wrapper = shallow(<Pocket {...initialProps} />);
    const CurrencyLabel = wrapper.find('CurrencyLabel');
    expect(CurrencyLabel.text()).toBe(initialProps.currencyLabel);
  });

  it('should have the right pocket amount', () => {
    const wrapper = shallow(<Pocket {...initialProps} />);
    const CurrencyLabel = wrapper.find('AmountLabel');
    expect(CurrencyLabel.contains(initialProps.pocketAmount)).toBe(true);
  });
});
