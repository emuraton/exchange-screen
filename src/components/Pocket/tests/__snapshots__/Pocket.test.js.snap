// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pocket /> should render correctly 1`] = `
<styled.div>
  <styled.div>
    <CurrencyLabel>
      Â£
    </CurrencyLabel>
    <AmountLabel>
      You have 
      EUR
      55.22
    </AmountLabel>
  </styled.div>
  <styled.div>
    <AmountInput
      autoFocus={true}
      onChange={[Function]}
      type="text"
      value=""
    />
  </styled.div>
</styled.div>
`;
